### ISSUE TITLE
[Torch] Support SmoothL1 Loss function in torch dialect

### ISSUE THREAD

**Issue by zahidwx** [llvm/torch-mlir#4249]
### SmoothL1 Loss function
https://docs.pytorch.org/docs/stable/generated/torch.nn.SmoothL1Loss.html

Smooth L1 loss function can be implemented either by:
1. decomposing it to torch primitive ops
2. using huber loss function

**Comment by zahidwx** [comment]
I would like to contribute to this issue.

**Comment by AiyyappanMR** [comment]
Hi, if no one's working on this yet, I'd like to pick it up.
